.github/workflows/ci-ui.yml

```diff
@@ -6,8 +6,17 @@ on:
         required: true
 
 jobs:
+  # Only run CI from unprotected branches (avoid CI for backporting)
+  check_branch:
+    name: Check origin branch
+    if: ${{ (github.head_ref != 'main') && (github.head_ref != 'develop') && !startsWith(github.head_ref, 'release') }}
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo Should run CI based on origin branch? - ${{ steps.should_run.outputs.should_run }}
```

Comment:
isn't the step `should_run` missing?

Reply:
Yeah, I removed the steps to leave only the conditional on line 12.  
That way if this job skips, all other jobs will skip as well.

Reply:
Yeah, but you are using it to print in this step `${{ steps.should_run.outputs.should_run }}`

Reply:
oh crap, you're right.
lemme fix it

Reply:
There, moved the conditional to the dependencies job since all others needs this job.  
Thank you Marciooo!